lib/kaffe/consumer_group/group_manager.ex

```diff
@@ -116,13 +128,42 @@ defmodule Kaffe.GroupManager do
   ## ==========================================================================
   ## Helpers
   ## ==========================================================================
+
   defp subscribe_to_topics(state, topics) do
-    for topic <- topics do
-      Logger.debug("Starting group member for topic: #{topic}")
-      {:ok, _pid} = subscribe_to_topic(state, topic)
+    Logger.debug("Starting group members for the following topics: #{inspect(topics)}")
+
+    retry with: exponential_backoff() |> expiry(client_down_retry_expire()),
+          rescue_only: [Kaffe.GroupManager.ClientDownException] do
+      Enum.each(topics, fn topic ->
+        case subscribe_to_topic(state, topic) do
+          {:ok, _pid} ->
+            Logger.debug("Started group member for topic: #{topic}")
+            :ok
+
+          error ->
+            Logger.debug("Starting group member for #{topic} failed, attempting retry with exponential backoff")
```

Comment:
Would it make sense to have the `attempting retry` part of the logging only happen if `do_a_retry?` comes back as true?

Reply:
Yes, it depends a bit on how you interpret it. The way I thought of it was attempting a retry depending on the retry settings and if the error is rescue-able. If the error is not rescue-able you will see this message only once, and it will then directly go to the else block which prints the error message which should explain why attempt failed.

It is also possible to limit this message only to rescue-able errors, thus letting the attempt only depend on the retry settings. I don't have a real preference for any of these two approaches. Considering that you can [purge log messages](https://hexdocs.pm/logger/1.11.3/Logger.html#module-application-configuration) of certain levels like `:debug` at compile time for prod and release environments I think that in the end it won't make much difference which approach we pick.